### ISSUE TITLE
Notice FeaturesController::get_compatible_plugins_for_feature was called incorrectly in tests

### ISSUE THREAD

**Issue by Chouby** [woothemes/woocommerce#41003]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

When running WooCommerce PHPUnit tests against the trunk branch, notices are emitted during the test bootstrap.
This issue doesn't exist when running the same test against the stable version 8.2.1

### Expected behavior

I expect no notice to be emitted when running tests.

### Actual behavior

A notice is emitted several times, I reformatted it to make it readable:
```
PHP Notice:  Function FeaturesController::get_compatible_plugins_for_feature was called incorrectly. FeaturesController::get_compatible_plugins_for_feature should not be called before the woocommerce_init action. Backtrace: include('/dev/git/others/wc-2/plugins/woocommerce/vendor/phpunit/phpunit/phpunit')
PHPUnit\TextUI\Command::main
PHPUnit\TextUI\Command-&gt;run
PHPUnit\TextUI\Command-&gt;handleArguments
PHPUnit\TextUI\Command-&gt;handleBootstrap
PHPUnit\Util\FileLoader::checkAndLoad
PHPUnit\Util\FileLoader::load
include_once('/dev/git/woocommerce/plugins/woocommerce/tests/legacy/bootstrap.php')
WC_Unit_Tests_Bootstrap::instance
WC_Unit_Tests_Bootstrap-&gt;__construct
require_once('/tmp/wordpress-tests-lib/includes/bootstrap.php')
require_once('wp-settings.php')
do_action('setup_theme')
WP_Hook-&gt;do_action
WP_Hook-&gt;apply_filters
WC_Unit_Tests_Bootstrap-&gt;install_wc
WC_Install::install
WC_Install::create_options
WC_Settings_Page-&gt;get_settings
WC_Settings_Page-&gt;get_settings_for_section
apply_filters('woocommerce_get_settings_advanced')
WP_Hook-&gt;apply_filters
Automattic\WooCommerce\Internal\Features\FeaturesController-&gt;__call
call_user_func_array
Automattic\WooCommerce\Internal\Features\FeaturesController-&gt;add_feature_settings
array_map
Automattic\WooCommerce\Internal\Features\FeaturesController-&gt;Automattic\WooCommerce\Internal\Features\{closure}
call_user_func
Automattic\WooCommerce\Internal\DataStores\Orders\CustomOrdersTableController-&gt;Automattic\WooCommerce\Internal\DataStores\Orders\{closure}
Automattic\WooCommerce\Internal\Features\FeaturesController-&gt;get_compatible_plugins_for_feature
Automattic\WooCommerce\Internal\Features\FeaturesController-&gt;verify_did_woocommerce_init
Automattic\WooCommerce\Proxies\LegacyProxy-&gt;call_function
call_user_func_array
wc_doing_it_wrong Please see <a>Debugging in WordPress</a> for more information. (This message was added in version 7.0.) in /dev/git/polylang/tmp/wordpress/wp-includes/functions.php on line 6031
```

### Steps to reproduce

1. Clone this repo
2. Go to subdirectory `plugins/woocommerce`
3. `composer update`
4. `php bin/generate-feature-config.php`
5. Run a few tests, for ex with  `vendor/bin/phpunit --filter WC_Tests_Product_Data_Store`

### WordPress Environment

WP latest
PHP 8.0.28
PHPUnit 9.6.13

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by coreymckrill** [comment]
Switching the team label on this to Mothra since it's related to WC Admin. Is the `bin/generate-feature-config.php` script still relevant?

**Comment by kidunot89** [comment]
@coreymckrill Has there been any progress on this issue. Our [CI](https://github.com/wp-graphql/wp-graphql-woocommerce/actions/runs/7137209246/job/19436853459#step:5:1235) is currently being broken by this issue when Codeception attempts to open activate WooCommerce

**Comment by coreymckrill** [comment]
@kidunot89 I took a fresh look at this, and I actually think this should be addressed by #41434, which went out just a few days ago in v8.4.0. Is your CI running with 8.4.0 yet?

**Comment by kidunot89** [comment]
@coreymckrill I can confirm it is working in v8.4.0, thanks for the speedy response.

**Comment by coreymckrill** [comment]
Excellent! I will close this as fixed by #41434
